my_enumerable.rb

```diff
@@ -0,0 +1,124 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    idx = to_a
+    idx.size.times do |index|
+      yield idx[index]
+    end
+
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    idx = to_a
+    idx.size.times do |index|
+      yield idx[index], index
+    end
+
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    select = []
+    my_each do |i|
+      select << i if yield(i)
+      select
+    end
+  end
```

Comment:
- [ ] `my_select` should return an array containing all of the elements for which the given block returns true when called on an array
```ruby
array = [1,2,3,4,56,6,7,53,23,45,1]
block =proc { |num| num < (1 + 9) / 2 }
array.my_select(&block ) === array.select(&block )  => false
```

- [ ] `my_select` should return an array containing all of the elements for which the given block returns true when called on a range

```ruby
range = Range.new(5,50)
block =proc { |num| num < (1 + 9) / 2 }
range.my_select(&block ) === range.select(&block )  => false
```